### PR TITLE
Build and launch the daemon using docker-compose

### PR DESCRIPTION
**Pull request by tonsV2**
This commit would allow developers to get quickly up and running. Simply do a `docker-compose up daemon` and now any changes made to the files under web/ will be visible upon a browser refresh.

### CODE REVIEW

File: daemon/settings.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    "queue-stalled-minutes": 30,
+    "ratio-limit": 2,
+    "ratio-limit-enabled": false,
+    "rename-partial-files": true,
+    "rpc-authentication-required": false,
+    "rpc-bind-address": "0.0.0.0",
+    "rpc-enabled": true,
+    "rpc-host-whitelist": "",
+    "rpc-host-whitelist-enabled": true,
+    "rpc-password": "{21707926375be5a45622a7d79d8bec1c05de7600ewNeeuls",
```
Comment (qdii) [inline daemon/settings.json:48]: Where does this password come from?

Would there be a way to leave password out, so that the user has to define them? default passwords tend to get hacked

Comment (tonsV2) [reply daemon/settings.json:48]: The password is `admin` since this is for development purposes.

A script could be written which updates or generates the settings.json file on the fly. But a different approached might be to implement a different image for production purposes. The main difference would be that it should include `web/`.
I'd love to implement that and if that's of interest we could easily configure a hook on Docker Hub which would trigger a build every time a push is made to Github.

It's worth noting that this image also runs as root which wouldn't be ideal for a production environment. I haven't experimented with this since my aim here was development rather than production.

**Comment by ckerr** [comment]
https://github.com/transmission/transmission/issues/750#issuecomment-1022849212

tldr: I'm happy for someone else to maintainand support a Docker image, but the project doesn't have the bandwidth to support a Docker image so I don't think it makes sense to include this in the repo.
